src/tests/nativeaot/SmokeTests/DynamicGenerics/DynamicGenerics.main.cs

```diff
@@ -165,7 +165,7 @@ public static int Main(string[] args)
         if (passed && CoreFXTestLibrary.Internal.Runner.NumPassedTests > 0)
         {
             CoreFXTestLibrary.Logger.LogInformation("All tests PASSED.");
-            return 100;
```

Comment:
Using `100` to indicate success is a convention used by all tests under src/tests. Changing this convention in the wasm branch is going to be perpetual conflict with upstream. Is this change really necessary?

Reply:
Initially, I followed @akoeplinger 's [advice](https://discord.com/channels/732297728826277939/732297825731215521/1253462821849792552) to print the exit code to standard out, but then @SingleAccretion [advised me](https://discord.com/channels/732297728826277939/732297825731215521/1255222443161948212) to change the codes instead.  I'm happy to do whatever you all think is best here.

For context: the reason we need to do _something_ special for WASIp2 here is https://github.com/WebAssembly/wasi-cli/issues/11

Reply:
+1 for xharness solution rather than editing `100` in too many places

Reply:
I do not see how you can avoid modifying the tests that are `Main`-based. There is no code to be hooked there (unless you start thinking of some really involved workaround like making everything use `CustomMain`).

If we need to modify the tests regardless, the simplest thing is to change the exit code.

(Actually, I wanted to probe grounds for doing this upstream as well, but [it may be problematic](https://discord.com/channels/143867839282020352/1141126727028985877/1255224816118534305).)

Reply:
could we just modify the native C main to printf exit code ? 
Perhaps only when the application is build with some flags on, so that production applications don't so it.

Reply:
> could we just modify the native C main to printf exit code ?
Perhaps only when the application is build with some flags on, so that production applications don't so it.

I have not seen a precedent for adding such test hooks into production code.

It wouldn't work for cases that exit abnormally (`Environment.Exit`) - we have one such test.

Reply:
> Especially for NAOT the application is recompiled every time, so #ifdef seems OK to me. For C# exit, we could have hook in PAL

The bootstrapper, which is where the native `main` is defined, is only compiled once, as part of the runtime build. I agree it is technically possible to make it work (e. g. via clever use of weak symbols, or compiler intrinsics, or something else).

My point is that it would not be to the scale of the problem, which is just running these smoke tests. If we don't convert all tests to use the zero exit code (unlikely), or don't wait enough for WASI P2.1 to add support for exit codes (possible but not terribly likely), then, when the question of how to run _all_ of the runtime tests comes up - which will be upstream, it will be solved in some more involved manner. At that time, the same solution, if technically possible, will be adopted downstream too.